api/src/services/server.ts

```diff
@@ -308,8 +308,10 @@ export class ServerService {
 
 				try {
 					await disk.write(`health-${checkID}`, Readable.from(['check']));
-					await disk.read(`health-${checkID}`);
-					await disk.delete(`health-${checkID}`);
+					const fileStream = await disk.read(`health-${checkID}`);
+					fileStream.on('data', async () => {
```

Comment:
Should this be on `finish` to be safe?

Reply:
Just tested and both `finish` or `data` work here and don't result in crashing healthcheck. Can update it to `finish`.

The idea is that it's waiting to make sure there's a valid read happening before we attempt to delete.

Reply:
We don't actually read the file, so neither `finish` nor `end` are ever getting called.
But it's fine to delete the file as soon as it has been opened (`data` signal).
I did quite a lot stress tests with the `data` signal and it has not failed once.

Reply:
> Yeah the `.on` is what matters here, testing with `data`, finish`and`end` all work the same and neither trigger crashes.

The test does work but the file is not getting deleted 🙃

Reply:
Didn’t see non-deleted files on local testing but can put it back to `data` for more  consistency then.

Reply:
> Didn’t see non-deleted files on local testing but can put it back to `data` for more consistency then.

Okay, that's interesting! I don't see how the `end` event should be called since we don't read from the stream. Besides there is no `finish` event on readable streams.

Thanks for putting it back to `data`!

One additional thing we could do to make it a bit safer though, is to destroy the stream before deleting the file:
```suggestion
					fileStream.on('data', async () => {
						fileStream.destroy();
```